src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -103,14 +103,24 @@ internal void SetRequestLine(string req)
             try
             {
                 _version = new Version(parts[2].Substring(5));
-                if (_version.Major < 1)
-                    throw new Exception();
             }
             catch
             {
                 _context.ErrorMessage = "Invalid request line (version).";
                 return;
             }
+
+            if (_version.Major < 1)
+            {
+                _context.ErrorMessage = "Invalid request line (version).";
+                return;
+            }
+            if (_version != HttpVersion.Version10 && _version != HttpVersion.Version11)
```

Comment:
> E.g 1.2, 1.3
There is no such thing as those versions.  There is only HTTP/1.0, HTTP/1.1 and HTTP/2.0.  Although I don't think HttpListener supports HTTP/2.0 at all.  And it is a transport protocol anyways for which HTTP/1.1 is carried over on it.